core/src/main/java/org/pentaho/di/connections/ConnectionManager.java

```diff
@@ -282,45 +349,49 @@ private List<String> getNames( IMetaStore metaStore, ConnectionProvider<? extend
    * @param provider A provider
    * @return A list of named connection names
    */
-  private List<String> getNames( ConnectionProvider<? extends ConnectionDetails> provider ) {
+  private List<String> loadNames( ConnectionProvider<? extends ConnectionDetails> provider ) {
     if ( metaStoreSupplier == null || metaStoreSupplier.get() == null ) {
       return Collections.emptyList();
     }
     return getNames( metaStoreSupplier.get(), provider );
   }
 
+  /**
+   * Get the names of named connections by provider from default meta store
+   *
+   * @param provider A provider
+   * @return A list of named connection names
+   */
+  private List<String> getNames( ConnectionProvider<? extends ConnectionDetails> provider ) {
+    initialize();
+    return namesByConnectionProvider.get( provider.getName() );
+  }
+
   /**
    * Get the names of named connections by provider from specified meta store
    *
    * @param metaStore  A meta store
-   * @param clearCache Whether or not to clear cache
+   * @param clearCache Whether or not to clear cache. (UNUSED)
```

Comment:
This will make it confusing that caller is allowed to pass boolean to clear the cache but it is not doing anything. 
Should we identify the cases where this method is called and do we want to clear the two hashmap

Reply:
Yeah, this refers to the old "nameCache" which is irrelevant with the whole thing being in memory, now. I would have removed the APIs that used this entirely, but didn't want to risk breaking any plugin in another repo (or third party) that used it.